tasks/move-granules/index.js

```diff
@@ -199,65 +389,169 @@ async function moveFileRequest(
+        ...moveParams,
+        file,
+        duplicateHandling,
+        markDuplicates: !isCmrFile,
+      }))
+  );
+}
+
 /**
  * Move all files in a collection of granules from staging location to final location,
  * and update granule files to include renamed files if any.
  *
- * @param {Object} granulesObject - an object of the granules where the key is the granuleId
- * @param {string} sourceBucket - source bucket location of files
- * @param {string} duplicateHandling - how to handle duplicate files
- * @param {number} s3MultipartChunksizeMb - S3 multipart upload chunk size in MB
- * @returns {Object} the object with updated granules
+ * This function processes all the granules and moves their files to the target locations,
+ * handling CMR files and regular files appropriately and updating granule metadata.
+ *
+ * @param {object} params - Move parameters
+ * @param {Collection} params.configCollection - Collection configuration
+ * @param {GranulesObject} params.granulesObject - an object of granules where key is granuleId
+ * @param {string} params.sourceBucket - source bucket location of files
+ * @param {DuplicateHandling} params.duplicateHandling - how to handle duplicate files
+ * @param {number} [params.s3MultipartChunksizeMb] - S3 multipart upload chunk size in MB
+ * @param {boolean} [params.checkCrossCollectionCollisions=true] - Whether to check
+ * for cross-collection collisions
+ * @param {object} [params.testOverrides={}] - Test overrides
+ * @returns {Promise<GranulesOutputObject>} the object with updated granules
  */
-async function moveFilesForAllGranules(
+async function moveFilesForAllGranules({
+  configCollection,
   granulesObject,
   sourceBucket,
   duplicateHandling,
-  s3MultipartChunksizeMb
-) {
+  s3MultipartChunksizeMb,
+  checkCrossCollectionCollisions = true,
+  testOverrides = {},
+}) {
   const moveFileRequests = Object.keys(granulesObject).map(async (granuleKey) => {
-    const granule = granulesObject[granuleKey];
-    const filesToMove = granule.files.filter((file) => !isCMRFile(file));
-    const cmrFiles = granule.files.filter((file) => isCMRFile(file));
-    const filesMoved = await Promise.all(
-      filesToMove.map((file) =>
-        moveFileRequest(file, sourceBucket, duplicateHandling, true, s3MultipartChunksizeMb))
-    );
-    const cmrFilesMoved = await Promise.all(
-      cmrFiles.map(
-        (file) => moveFileRequest(file, sourceBucket, 'replace', false)
-      )
+    const filesToMove = granulesObject[granuleKey].files.filter((file) => !isCMRFile(file));
+    const cmrFiles = granulesObject[granuleKey].files.filter((file) => isCMRFile(file));
+
+    const granuleCollectionId = determineGranuleCollectionId(
+      granulesObject[granuleKey],
+      configCollection
     );
+    if (!granuleCollectionId) {
+      throw new ValidationError(`Unable to determine collection ID for granule ${granuleKey}`);
+    }
+
+    /** @type {MoveGranulesGranuleOptionalFilesFields} */
```

Comment:
This cast/assignment is needed to widen the output type.   The input requires fields, but once used for filtering are not needed again.     The lambda mutates granule here rather than reassign for memory usage reasons.